grid/project-management/tasks/maps/atlassian-cloud.suma

```diff
@@ -0,0 +1,348 @@
+          issuetype: {
+            name: "Task"
+          },
+          project: {
+            id: input.projectIds[0]
+          },
+          summary: input.title,
+          // https://developer.atlassian.com/cloud/jira/platform/apis/document/structure/
+          description: input.description ? {
+            "version": 1,
+            "type": "doc",
+            "content": [
+              {
+                "type": "paragraph",
+                "content": [
+                  {
+                    "type": "text",
+                    "text": input.description
+                  }
+                ]
+              }
+            ]
+          } : undefined,
+          parent: input.parent ? {
+            key: input.parent
+          } : undefined,
+          assignee: input.assignee ? {
+            id: input.assignee
+          } : undefined
+        }
+      }
+    }
+  
+    response 201 "application/json" {
+      return map result {
+        taskId = body.id
+        url = body.self
+      }
+    }
+
+    response "application/json" {
+      error = call HandleError(body = body, statusCode = statusCode)
+      return map error error
+    }
+  
+    response {
+      return map error {
+        kind = "UnknownError",
+        title = "Unknown Error",
+        detail = `Status code: ${statusCode}, Received body: ${JSON.stringify(body)}`
+      }
+    }
+  }
+}
+
+map ListTasks {
+  http GET "/ex/jira/{parameters.CLOUD_ID}/rest/api/3/search" {
+    security "bearerToken"
+  
+    request {
+      query {
+        jql = `project=${input.projectId}`
+        startAt = input.page
+      }
+    }
+  
+    response 200 "application/json" {
+      processed = call ProcessIssues(issues=body.issues)
+      return map result {
+        tasks = processed.tasks
+        nextPage = String((parseInt(input.page || "0")) + processed.tasks.length)
+      }
+    }
+  
+    response "application/json" {
+      error = call HandleError(body = body, statusCode = statusCode)
+      return map error error
+    }
+  
+    response {
+      return map error {
+        kind = "UnknownError",
+        title = "Unknown Error",
+        detail = `Status code: ${statusCode}, Received body: ${JSON.stringify(body)}`
+      }
+    }
+  }
+}
+
+map ReadTask { 
+  http GET "/ex/jira/{parameters.CLOUD_ID}/rest/api/3/issue/{input.taskId}" {
+    security "bearerToken"
+  
+    response 200 "application/json" {
+      processed = call ProcessIssues(issues=[body])
+      return map result processed.tasks[0]
+    }
+  
+    response "application/json" {
+      error = call HandleError(body = body, statusCode = statusCode)
+      return map error error
+    }
+  
+    response {
+      return map error {
+        kind = "UnknownError",
+        title = "Unknown Error",
+        detail = `Status code: ${statusCode}, Received body: ${JSON.stringify(body)}`
+      }
+    }
+  }
+}
+
+map UpdateTask { 
+  return map error if ((input.projectIds || []).length > 1) {
```

Comment:
nitpick: this validation can be moved to operation and shared

Reply:
The checks in `CreateTask` and `UpdateTask` are different.